Sources/Core/ObjCADTRenderer.swift

```diff
@@ -102,6 +102,20 @@ struct ObjCADTRenderer: ObjCFileRenderer {
         }
     }
 
+    func renderDictionaryRepresentation() -> ObjCIR.Method {
+            return ObjCIR.method("- (NSDictionary*)dictionaryRepresentation") {
+                [
+                    ObjCIR.switchStmt("self.internalType") {
+                        self.dataTypes.enumerated().map { (index, schema) -> ObjCIR.SwitchCase in
+                            return ObjCIR.caseStmt(self.internalTypeEnumName + ObjCADTRenderer.objectName(schema)) {[
+                                    ObjCIR.stmt("return [[NSDictionary alloc]initWithDictionary:[self.value\(index) dictionaryRepresentation]]")
```

Comment:
"One of" can be any data type (URL, Date, Int, etc) so we can't just use `dictionaryRepresentation` for all values. We need to have the ADT method perform the same logic in the `renderAddObject` implementation you have already.